src/metax_api/api/oaipmh/base/metax_oai_server.py

```diff
@@ -39,15 +39,28 @@ def _get_filtered_records(self, set, cursor, batch_size, from_=None, until=None)
         elif until:
             query_set = CatalogRecord.objects.filter(date_modified__lte=until)
 
-        if set and set != 'urnresolver':
+        if set and set != "urnresolver":
             query_set = query_set.filter(data_catalog__catalog_json__identifier__in=settings.OAI['SET_MAPPINGS'][set])
         else:
             query_set = query_set.filter(data_catalog__catalog_json__identifier__in=self._get_default_set_filter())
         return query_set[cursor:batch_size]
 
-    def _get_oai_dc_urnresolver_metadata(self, source_identifier, target_identifier):
+    def _get_oai_dc_urnresolver_metadata(self, record):
+        """
+        Preferred identifier is added only for ida and att catalog records
+        other identifiers are added for all.
+        """
+        identifiers = []
+        identifiers.append(settings.OAI['ETSIN_URL_TEMPLATE'] % record.urn_identifier)
+        if record.catalog_versions_datasets():
+            identifiers.append(record.urn_identifier)
```

Comment:
Was the plan to export (unique) preferred_identifiers only, and no urn_identifiers (metadata_version_identifier) ? Or is this code relevant for that part

Reply:
This is implemented according to the current versioning scheme. I will change it when urn_identifier field is removed.

Reply:
I mean in general. preferred_identifier should be the one that is exported?

Reply:
Hmmm...You're right. 
What is being exported is one url and n urns. And urns should come from preferred identifier and other identifiers. I'm going to change that piece of code.